databricks/koalas/indexes.py

```diff
@@ -1484,6 +1484,16 @@ def delete(self, loc):
                     "index {} is out of bounds for axis 0 with size {}".format(loc, length)
                 )
             loc = [loc]
+        # pandas with numpy>=1.19.0 raises Exception if the item in the list exceeds the length.
+        # Because the behavior of `np.delete` has been changed from numpy 1.19.0
+        # which pandas uses directly.
+        elif LooseVersion(np.__version__) >= LooseVersion("1.19.0"):
+            for index in loc:
+                if abs(index) >= length:
```

Comment:
Btw, this is not right?

```py
>>> pidx = pd.Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10])
>>> len(pidx)
12
>>> pidx.delete(-12)
Int64Index([10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
>>> pidx.delete([-12])
Int64Index([10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
>>> pidx.delete(-13)
Traceback (most recent call last):
...
IndexError: index -13 is out of bounds for axis 0 with size 12
>>> pidx.delete([-13])
Traceback (most recent call last):
...
IndexError: index -13 is out of bounds for axis 0 with size 12
```

whereas:

```py
>>> kidx = ks.from_pandas(pidx)
>>> kidx.delete(-12)
Traceback (most recent call last):
...
IndexError: index -12 is out of bounds for axis 0 with size 12
>>> kidx.delete([-12])
Traceback (most recent call last):
...
IndexError: index -12 is out of bounds for axis 0 with size 12
```

Reply:
Hey @itholic, let's make sure we check the corner cases before you make a PR.

Reply:
Thanks, @ueshin @HyukjinKwon .
I'll make a follow-up soon and will more careful before open up the PR.